pkg/pillar/pubsub/pubsub.go

```diff
@@ -7,39 +7,28 @@
 package pubsub
 
 import (
-	"encoding/json"
-	"errors"
-	"fmt"
-	"io/ioutil"
-	"net"
-	"os"
-	"path"
-	"strings"
-	"time"
-
-	"github.com/google/go-cmp/cmp"
 	log "github.com/sirupsen/logrus"
 )
 
-// Protocol over AF_UNIX or other IPC mechanism
-// "request" from client after connect to sanity check subject.
-// Server sends the other messages; "update" for initial values.
-// "complete" once all initial keys/values in collection have been sent.
-// "restarted" if/when pub.km.restarted is set.
-// Ongoing we send "update" and "delete" messages.
-// They keys and values are base64-encoded since they might contain spaces.
-// We include typeName after command word for sanity checks.
-// Hence the message format is
-//	"request" topic
-//	"hello"  topic
-//	"update" topic key json-val
-//	"delete" topic key
-//	"complete" topic (aka synchronized)
-//	"restarted" topic
-
-// Used locally by each serverConnection goroutine to track updates
-// to send.
-type localCollection map[string]interface{}
+// SubHandler is a generic handler to handle create, modify and delete
+// Usage:
+//  s1 := pubsub.Subscribe("foo", fooStruct{}, true, &myctx)
+// Or
+//  s1 := pubsub.Subscribe("foo", fooStruct{}, false, &myctx)
+//  s1.ModifyHandler = func(...), // Optional
+//  s1.DeleteHandler = func(...), // Optional
+//  s1.RestartHandler = func(...), // Optional
+//  [ Initialize myctx ]
+//  s1.Activate()
+//  ...
+//  select {
+//     case change := <- s1.C:
+//         s1.ProcessChange(change, ctx)
+//  }
+type SubHandler func(ctx interface{}, key string, status interface{})
```

Comment:
Is this the default handler that callers can provide to when Subscribing - if they don't want to provide these various handlers?

Reply:
This just defines the type; it is used when actually defining the handlers on `Subscription`.